qiita_pet/handlers/rest/study_association.py

```diff
@@ -0,0 +1,224 @@
+
+    return sorted(all_artifact_objects, key=lambda artifact: artifact.id)
+
+
+def _set_artifacts(template_payload, prep_template):
+    """Set artifact information specific to a prep"""
+    prep_artifacts = []
+
+    if prep_template.artifact is None:
+        basal_id = None
+    else:
+        basal_id = prep_template.artifact.id
+
+    for artifact in _get_artifacts(prep_template):
+        _set_artifact(prep_artifacts, artifact, basal_id)
+    template_payload[PREP_ARTIFACTS] = prep_artifacts
+
+
+def _set_artifact(prep_artifacts, artifact, basal_id):
+    """Set artifact specific information"""
+    artifact_payload = {}
+    artifact_payload[ARTIFACT_ID] = artifact.id
+
+    # Prep uses .status, artifact uses .visibility
+    # favoring .status as visibility implies a UI
+    artifact_payload[ARTIFACT_STATUS] = artifact.visibility
+
+    parents = [parent.id for parent in artifact.parents]
+    artifact_payload[ARTIFACT_PARENT_IDS] = parents if parents else None
+    artifact_payload[ARTIFACT_BASAL_ID] = basal_id
+
+    _set_artifact_processing(artifact_payload, artifact)
+    _set_artifact_filepaths(artifact_payload, artifact)
+
+    prep_artifacts.append(artifact_payload)
+
+
+def _set_artifact_processing(artifact_payload, artifact):
+    """Set processing parameter information associated with an artifact"""
+    processing_parameters = artifact.processing_parameters
+    if processing_parameters is None:
+        artifact_processing_id = None
+        artifact_processing_name = None
+        artifact_processing_arguments = None
+    else:
+        command = processing_parameters.command
+        artifact_processing_id = command.id
+        artifact_processing_name = command.name
+        artifact_processing_arguments = processing_parameters.values
+
+    artifact_payload[ARTIFACT_PROCESSING_ID] = artifact_processing_id
+    artifact_payload[ARTIFACT_PROCESSING_NAME] = artifact_processing_name
+    artifact_payload[ARTIFACT_PROCESSING_ARGUMENTS] = artifact_processing_arguments
+
+
+def _set_artifact_filepaths(artifact_payload, artifact):
+    """Set filepath information associated with an artifact"""
+    artifact_filepaths = []
+    for filepath_data in artifact.filepaths:
+        local_payload = {}
+        local_payload[ARTIFACT_FILEPATH] = filepath_data['fp']
+        local_payload[ARTIFACT_FILEPATH_ID] = filepath_data['fp_id']
+        local_payload[ARTIFACT_FILEPATH_TYPE] = filepath_data['fp_type']
+        artifact_filepaths.append(local_payload)
+
+    # the test study includes an artifact which does not have filepaths
+    if len(artifact_filepaths) == 0:
+        artifact_filepaths = None
+
+    artifact_payload[ARTIFACT_FILEPATHS] = artifact_filepaths
+
+
+class StudyAssociationHandler(RESTHandler):
+    @authenticate_oauth
+    def get(self, study_id):
+        study = self.safe_get_study(study_id)
+        if study is None:
+            return
+
+        # schema:
```

Comment:
Nice! I think adding something like this in the endpoint docs, here: https://github.com/qiita-spots/qiita/blob/master/qiita_pet/support_files/doc/source/dev/rest.rst would also be good.